plumbing/transport/fetch_pack.go

```diff
@@ -128,6 +128,28 @@ func (r *UploadPackRequest) Have(h ...plumbing.Hash) {
 	r.Haves = append(r.Haves, h...)
 }
 
+func (r *UploadPackRequest) IsEmpty() bool {
+	if len(r.Haves) < len(r.Wants) {
+		return false
+	}
```

Comment:
From https://github.com/git/git/blob/master/Documentation/technical/pack-protocol.txt :
> Clients MUST NOT mention an obj-id in a 'want' command which did not appear in the response obtained through ref discovery.

As the `UploadPackRequest.Want` does not enforce this requisite we cannot count on it. This means we cannot relay on lines 132-134 for an early return.